### PR TITLE
Update API docs for "HostConfig.Binds"

### LINKED ISSUE #12128: API doc for container create "Binds" argument incorrect, and error message confusing
The api doc for the [`Binds` argument](https://docs.docker.com/reference/api/docker_remote_api_v1.17/#create-a-container) is incorrect.  When creating a new volume, the volume should not appear in "Binds".  The [implementation](https://github.com/docker/docker/blob/master/daemon/volumes.go#L225) explicity checks for a string with two or three components separated by ":".

Moreover, the resulting error message is "Invalid volume specification", which is confusing, as it doesn't mention the `Binds` argument at all.

### PR DESCRIPTION
**Pull request by thaJeztah**
fixes https://github.com/docker/docker/issues/12128

ping @cpuguy83 PTAL

### CODE REVIEW

File: docs/reference/api/docker_remote_api_v1.21.md [hunk 1]
```diff
@@ -248,11 +248,17 @@ Create a container
...
-           + `volume_name:container_path` to bind-mount a volume managed by a volume plugin into the container.
-           + `volume_name:container_path:ro` to make the bind mount read-only inside the container.
+           + `host-src:container-dest` to bind-mount a host path into the
+             container. Both `host-src`, and `container-dest` must be an
+             _absolute_ path.
+           + `host-src:container-dest:ro` to make the bind-mount read-only
+             inside the container. Both `host-src`, and `container-dest` must be
+             an _absolute_ path.
+           + `volume-name:container-dest` to bind-mount a volume managed by a
+             volume plugin into the container. `container-dest` must be an
```
Comment (cpuguy83) [inline docs/reference/api/docker_remote_api_v1.21.md]: s/plugin/driver

File: docs/reference/api/docker_remote_api_v1.22.md [hunk 2]
```diff
@@ -351,10 +351,17 @@ Create a container
...
-           + `volume_name:container_path` to bind-mount a volume managed by a volume plugin into the container.
-           + `volume_name:container_path:ro` to make the bind mount read-only inside the container.
+           + `host-src:container-dest` to bind-mount a host path into the
+             container. Both `host-src`, and `container-dest` must be an
+             _absolute_ path.
+           + `host-src:container-dest:ro` to make the bind-mount read-only
+             inside the container. Both `host-src`, and `container-dest` must be
+             an _absolute_ path.
+           + `volume-name:container-dest` to bind-mount a volume managed by a
+             volume plugin into the container. `container-dest` must be an
```
Comment (cpuguy83) [inline docs/reference/api/docker_remote_api_v1.22.md]: s/plugin/driver

File: docs/reference/api/docker_remote_api_v1.23.md [hunk 3]
```diff
@@ -374,10 +374,17 @@ Create a container
...
-           + `volume_name:container_path` to bind-mount a volume managed by a volume plugin into the container.
-           + `volume_name:container_path:ro` to make the bind mount read-only inside the container.
+           + `host-src:container-dest` to bind-mount a host path into the
+             container. Both `host-src`, and `container-dest` must be an
+             _absolute_ path.
+           + `host-src:container-dest:ro` to make the bind-mount read-only
+             inside the container. Both `host-src`, and `container-dest` must be
+             an _absolute_ path.
+           + `volume-name:container-dest` to bind-mount a volume managed by a
+             volume plugin into the container. `container-dest` must be an
```
Comment (cpuguy83) [inline docs/reference/api/docker_remote_api_v1.23.md]: s/plugin/driver

File: docs/reference/api/docker_remote_api_v1.24.md [hunk 4]
```diff
@@ -391,10 +391,17 @@ Create a container
...
-           + `volume_name:container_path` to bind-mount a volume managed by a volume plugin into the container.
-           + `volume_name:container_path:ro` to make the bind mount read-only inside the container.
+           + `host-src:container-dest` to bind-mount a host path into the
+             container. Both `host-src`, and `container-dest` must be an
+             _absolute_ path.
+           + `host-src:container-dest:ro` to make the bind-mount read-only
+             inside the container. Both `host-src`, and `container-dest` must be
+             an _absolute_ path.
+           + `volume-name:container-dest` to bind-mount a volume managed by a
+             volume plugin into the container. `container-dest` must be an
```
Comment (cpuguy83) [inline docs/reference/api/docker_remote_api_v1.24.md]: s/plugin/driver

File: docs/reference/api/docker_remote_api_v1.25.md [hunk 5]
```diff
@@ -395,10 +395,17 @@ Create a container
...
-           + `volume_name:container_path` to bind-mount a volume managed by a volume plugin into the container.
-           + `volume_name:container_path:ro` to make the bind mount read-only inside the container.
+           + `host-src:container-dest` to bind-mount a host path into the
+             container. Both `host-src`, and `container-dest` must be an
+             _absolute_ path.
+           + `host-src:container-dest:ro` to make the bind-mount read-only
+             inside the container. Both `host-src`, and `container-dest` must be
+             an _absolute_ path.
+           + `volume-name:container-dest` to bind-mount a volume managed by a
+             volume plugin into the container. `container-dest` must be an
```
Comment (cpuguy83) [inline docs/reference/api/docker_remote_api_v1.25.md]: s/plugin/driver

**Review by cpuguy83 — CHANGES_REQUESTED** [review]

**Comment by thaJeztah** [comment]
@cpuguy83 updated, ptal

**Review by cpuguy83 — APPROVED** [review]
LGTM

**Comment by thaJeztah** [comment]
thanks! author LGTM

**Pull request merged**